fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -581,28 +598,198 @@ void basicGroupedPartitionedTest() {
             validateSegmentAndIndexIntegrity(COMPLEX_PARTITIONED, partition1Subspace, context, "_0.cfs");
             validateSegmentAndIndexIntegrity(COMPLEX_PARTITIONED, partition2Subspace, context, "_0.cfs");
         }
+    }
+
+    @Test
+    void repartitionGroupedTest() throws ExecutionException, InterruptedException, InvalidProtocolBufferException {
```

Comment:
We should probably add a test with a bunch of records with the same timestamp. Or note it as an outstanding issue.